drivers/overlay/peerdb.go

```diff
@@ -244,13 +237,12 @@ func (d *driver) peerDbUpdateSandbox(nid string) {
 
 		peerOps = append(peerOps, op)
 	}
-	pMap.Unlock()
 
 	for _, op := range peerOps {
 		op()
 	}
+	pMap.Unlock()
```

Comment:
@fcrisciani @mavenugo `peerDbWg.Add` is only in the sandbox init path. In the pre-swarm mode overlay using serf there is no network scoped joining of the cluster. So what used to happen was all the peer events for a network which doesn't have local tasks is stored in the peerDB. When the first container comes up on the network `peerDbUpdateSandbox` would program all those entries into the overlay network. If we didn't do this, till the next self pushpull there won't be connectivity to the peers. I think the only reason peerDbWg is needed is to make sure this one shot `peerDbUpdateSandbox` is done before we start processing the serf events. For ex: the serf could be for a delete of an existing endpoint that we received earlier and in the peerDB.

With the networkdb in swarm mode since the network gets recreated on a task run on the node and get all the events from the peers, I don't see a strong reason for using the peerDbWg. One option here is to add do make the peerDbWg calls (Add/Wait/Done) conditional and only for non-swarm mode so that we won't change the flow for the non-swarm mode and avoid the panic for the swarm mode.

For the non-swarm mode the panic is still possible. But there is very little usage for it now and we can take some time to fix it in a different approach (or may be we might even deprecate that mode) rather than completely removing that logic.